backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/AddVmCommand.java

```diff
@@ -843,6 +840,12 @@ protected boolean validate() {
         return true;
     }
 
+    private Map<Guid, List<DiskImage>> getStorageToDisksMap() {
+        return storageToDisksMap == null ?
+                ImagesHandler.buildStorageToDiskMap(getImagesToCheckDestinationStorageDomains(),
+                        diskInfoDestinationMap) : storageToDisksMap;
+    }
```

Comment:
that's better than it was before but then let's set `storageToDisksMap` if it's null and change all places that access `storageToDisksMap` to use this method

Reply:
Ack